frontend/js/src/pins/PinRecordingModal.tsx

```diff
@@ -28,8 +29,13 @@ export const maxBlurbContentLength = 280;
  */
 
 export default NiceModal.create(
-  ({ recordingToPin, onSuccessfulPin }: PinRecordingModalProps) => {
+  ({
+    recordingToPin,
+    onSuccessfulPin,
+    rowId,
+  }: PinRecordingModalProps) => {
     const modal = useModal();
+    const isUpdate = Boolean(rowId == null);
```

Comment:
I think this should be sufficient, since all numbers apart from 0 will return true:
```suggestion
    const isUpdate = Boolean(rowId);
```

In comparison, even now the code seems unreliable, as `rowId` should eitgher be a number or undefined, but not null.
If for some reason in the future rowId ends up being undefined instead of null or vice versa, the check could simply break.

I'd also point out more generally that `==` and `===` don't mean the same thing in javascript; you would probably want "strict equality" here

Reply:
Also it seems the `==` work with undefined as well as null, mentioned here: https://stackoverflow.com/a/2647888